lib/systems/default.nix

```diff
@@ -112,15 +131,24 @@ let
         else if final.isNone                  then "newlib"
         # TODO(@Ericson2314) think more about other operating systems
         else                                     "native/impure";
+
+      unwinderlib =
+        /**/ if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then "libunwind"
```

Comment:
Should Darwin be a distinct value? It uses the system unwinder, which is based on libunwind but not necessarily the upstream libunwind. If so, I’d make it something like `libunwind-system`. That should simplify the checks in LLVM for whether to link libunwind.